web/src/engine/main/src/keymanEngine.ts

```diff
@@ -190,6 +194,42 @@ export default class KeymanEngine<ContextManager extends ContextManagerBase, Har
 
     return report;
   }
+
+  private refreshModel() {
+    const kbd = this.contextManager.activeKeyboard;
+    const model = this.modelCache.modelForLanguage(kbd.metadata.langId);
+
+    if(this.processor.activeModel != model) {
+      if(this.processor.activeModel) {
+        this.processor.languageProcessor.unloadModel();
+      }
+
+      if(model) {
+        this.processor.languageProcessor.loadModel(model);
+      }
+    }
+  }
```

Comment:
Why is this change needed?

Reply:
The old "ModelManager" class from namespaced-code was both a "model cache" and "model activator".  These two roles are split by this PR.  The `ModelCache` class maintains the former, while `KeymanEngine` itself will handle the latter, with this method as the primary method to do exactly that.

Compare this with the old `ModelManager.onKeyboardChange` method.  This method is pretty much just a renamed version of that, but based in a different location (`KeymanEngine`) than before.